test/extensions/filters/listener/http_inspector/http_inspector_test.cc

```diff
@@ -485,6 +553,77 @@ TEST_F(HttpInspectorTest, MultipleReadsHttp1BadProtocol) {
   EXPECT_EQ(1, cfg_->stats().http_not_found_.value());
 }
 
+TEST_F(HttpInspectorTest, Http1WithLargeRequestLine) {
+  init();
+  absl::string_view method = "GET", http = "/index HTTP/1.0\r";
+  std::string spaces(Config::MAX_INSPECT_SIZE - method.length() - http.length(), ' ');
+  const std::string data = absl::StrCat(method, spaces, http);
+  {
+    InSequence s;
+
+    EXPECT_CALL(os_sys_calls_, recv(42, _, _, MSG_PEEK)).WillOnce(InvokeWithoutArgs([]() {
+      return Api::SysCallSizeResult{ssize_t(-1), EAGAIN};
+    }));
+
+    for (size_t i = 1; i <= 2; i++) {
```

Comment:
Was this change to address TSAN? You may want change the number of iteration based on configuration, we do this in other places like: https://github.com/envoyproxy/envoy/blob/master/test/extensions/filters/http/lua/lua_filter_test.cc#L1271

Reply:
Done